src/inet/TCPEndPoint.cpp

```diff
@@ -2147,6 +2133,64 @@ void TCPEndPoint::LwIPHandleError(void *arg, err_t lwipErr)
 
 #if WEAVE_SYSTEM_CONFIG_USE_SOCKETS
 
+INET_ERROR TCPEndPoint::BindSrcAddrFromIntf(IPAddressType addrType, InterfaceId intf)
+{
+    INET_ERROR err = INET_NO_ERROR;
+
+    // If we are trying to make a TCP connection over IPv4 and a 'specified target interface',
+    // then we bind the TCPEndPoint to an IPv4 address on that target interface and use that
+    // address as the source address for that connection.
+    IPAddress curAddr = IPAddress::Any;
+    InterfaceId curIntfId = INET_NULL_INTERFACEID;
+    bool ipAddrFound = false;
+
+    VerifyOrExit(State != kState_Bound, err = INET_ERROR_NOT_SUPPORTED);
+
+    for (InterfaceAddressIterator addrIter; addrIter.HasCurrent(); addrIter.Next())
+    {
+        curAddr = addrIter.GetAddress();
+        curIntfId = addrIter.GetInterface();
```

Comment:
Test for curIntfId == intf here?

Reply:
That's right. It's being redundantly being checked at both places.